### PR TITLE
Initial support for Azure pricing

### PR DESCRIPTION
**Pull request by aliscott**
This adds support for the following Azure resources:
* azurerm_linux_virtual_machine
  - Pricing: https://azure.microsoft.com/en-gb/pricing/details/virtual-machines/linux/
  - Terraform: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine
* azurerm_windows_virtual_machine
  - Pricing: https://azure.microsoft.com/en-gb/pricing/details/virtual-machines/windows/
  - Terraform: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine
  
 Without usage:
 ```
  Name                                                    Quantity  Unit                Monthly Cost

 azurerm_linux_virtual_machine.example_linux
 ├─ Instance usage (pay as you go, A2 v2)                     730  hours                     $66.43
 └─ os_disk
    ├─ Storage (E4)                                             1  months                     $2.40
    └─ Disk operations                            Cost depends on usage: $0.002 per 10k operations

 azurerm_windows_virtual_machine.example_windows
 ├─ Instance usage (hybrid benefit, D2a v4)                   730  hours                     $70.08
 └─ os_disk
    ├─ Storage (S4)                                             1  months                     $1.54
    └─ Disk operations                            Cost depends on usage: $0.0005 per 10k operations
```

With usage:
 ```
 azurerm_linux_virtual_machine.example_linux
 ├─ Instance usage (pay as you go, A2 v2)              730  hours                 $66.43
 └─ os_disk
    ├─ Storage (E4)                                      1  months                 $2.40
    └─ Disk operations                                 200  10k operations         $0.40

 azurerm_windows_virtual_machine.example_windows
 ├─ Instance usage (hybrid benefit, D2a v4)            730  hours                 $70.08
 └─ os_disk
    ├─ Storage (S4)                                      1  months                 $1.54
    └─ Disk operations                                 200  10k operations         $0.10

 PROJECT TOTAL                                                                   $580.95
```

### CODE REVIEW

File: infracost-usage-example.yml [hunk 1]
```diff
@@ -406,3 +406,13 @@ resource_usage:
...
+
+  #
+  # Terraform AzureRM resources
+  #
+  azurerm_linux_virtual_machine.my_linux_vm:
+    os_disk:
+      monthly_disk_operations: 2000000 # The number of disk operations (writes, reads, deletes) using a unit size of 256KiB
+  azurerm_windows_virtual_machine.my_windows_vm:
+    os_disk:
+      monthly_disk_operations: 2000000 # The number of disk operations (writes, reads, deletes) using a unit size of 256KiB
```
Comment (alikhajeh1) [inline infracost-usage-example.yml:418]:
```suggestion
      monthly_disk_operations: 2000000 # Number of disk operations (writes, reads, deletes) using a unit size of 256KiB.
  azurerm_windows_virtual_machine.my_windows_vm:
    os_disk:
      monthly_disk_operations: 2000000 # Number of disk operations (writes, reads, deletes) using a unit size of 256KiB.
```

File: internal/providers/terraform/azure/linux_virtual_machine.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+// todo: add devtest consumption
```
Comment (alikhajeh1) [inline internal/providers/terraform/azure/linux_virtual_machine.go:12]: Should this be mentioned in Notes below?

File: internal/providers/terraform/azure/virtual_machine.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	"github.com/shopspring/decimal"
+	log "github.com/sirupsen/logrus"
+	"github.com/tidwall/gjson"
+)
+
+var diskSizeMap = map[string][]struct {
+	Name string
+	Size int
+}{
+	"Standard_LRS": {
```
Comment (alikhajeh1) [inline internal/providers/terraform/azure/virtual_machine.go:17]:
```suggestion
        // The mapping is from https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types
	"Standard_LRS": {
```

File: internal/providers/terraform/azure/virtual_machine.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+		var opsQty *decimal.Decimal
+
+		if u != nil && u.Get("os_disk.monthly_disk_operations").Exists() {
+			opsQty = decimalPtr(decimal.NewFromInt(u.Get("os_disk.monthly_disk_operations").Int()).Div(decimal.NewFromInt(10000)))
+		}
+
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:            "Disk operations",
+			Unit:            "10k operations",
+			UnitMultiplier:  1,
```
Comment (alikhajeh1) [inline internal/providers/terraform/azure/virtual_machine.go:152]: Should this be like this?

```suggestion
			Unit:            "operations",
			UnitMultiplier:  10000,
```

File: internal/providers/terraform/azure/windows_virtual_machine.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+// todo: add devtest consumption
```
Comment (alikhajeh1) [inline internal/providers/terraform/azure/windows_virtual_machine.go:12]: Should this be mentioned in Notes below?

**Review by alikhajeh1 — APPROVED** [review]
Great to see Azure resources added :) Added my suggestions

File: internal/providers/terraform/azure/virtual_machine.go (hunk 4 above)
Comment (aliscott) [reply internal/providers/terraform/azure/virtual_machine.go:153]: The price comes through as per 10k, so can't be done that way.

File: internal/providers/terraform/azure/linux_virtual_machine.go (hunk 2 above)
Comment (aliscott) [reply internal/providers/terraform/azure/linux_virtual_machine.go:12]: Sorry, I should have removed this todo. Will update.

**Pull request merged**